ec2-dash/dash_sqlite.py

```diff
@@ -185,18 +190,30 @@ def push_to_s3():
                 most_recent_ride_to_csv(cursor)
                 user_info_to_csv(cursor)
                 push_to_s3()
-                create_new_current_ride_table(cursor, conn)
-                os.remove("ec2-dash/most_recent_ride.csv") # delete the old log csv
-                try:
-                    os.remove("ec2-dash/user_info.csv") # delete the old user csv
-                    create_user_info_table(cursor, conn, log_entry)
-                except:
-                    create_user_info_table(cursor, conn, log_entry)
+                
+                # delete expired csv files 
+                os.remove("ec2-dash/most_recent_ride.csv") 
+                os.remove("ec2-dash/user_info.csv") 
+                os.remove("ec2-dash/join_ids.csv") 
+
+                #regenerate the tables
+                recreate_current_ride_table(cursor, conn)
+                recreate_user_info_table(cursor, conn, log_entry)
+
+                # regenerate the join_ids and user_info csvs
+                ride_id = recreate_ride_id_from_datetime(log_entry)
+                ride_date = log_entry['date'] + " " + log_entry['time']
+                recreate_join_csv(log_entry['user_id'], ride_id)
+
+                # set new max_hr
+                # max_hr = calculate_max_heart_rate(log_entry['date_of_birth'])
                 
 
             # only adds to the database if it is a full entry      
             if len(log_entry) == 7:
-                add_entry_to_table(cursor, conn, log_entry)
+                add_entry_to_table(cursor, conn, log_entry, ride_id)
+                # if compare_hr_to_max_hr(log_entry['heart_rate'], max_hr) == True:
+                    # send_user_info_to_lambda(log_entry['heart_rate'], log_entry['duration'])
 
```

Comment:
Like the max hr calculation setup, could we pas this to the dashboard as part of the user info dict to include a visual representation of the max on the hr graph?

Reply:
I like this idea, shifted the code round and the user_info now includes max_hr as the last entry